Lib/test/test_tarfile.py

```diff
@@ -858,7 +859,10 @@ def test_read_through(self):
                                   "TarFile.extractfile() failed")
                     if not buf:
                         break
-
+    def test_read_with_extra_header(self):
```

Comment:
```suggestion

    @unittest.skipIf(zlib is None, "requires zlib")
    def test_read_with_extra_header(self):
```
WASI job fails because it is missing zlib. I'm not sure if this is the right approach, however.